src/mpi/coll/helper_fns.c

```diff
@@ -39,6 +39,8 @@ int MPIC_Probe(int source, int tag, MPI_Comm comm, MPI_Status * status)
   fn_exit:
     return mpi_errno;
   fn_fail:
+    if (mpi_errno == MPIX_ERR_NOREQ)
+        MPIR_ERR_SET(mpi_errno, MPI_ERR_OTHER, "**nomem");
```

Comment:
This is weird. Are we resetting the error class to `MPI_ERR_OTHER` here?

Reply:
For the collective helper functions, I just convert the non-catastrophic error to a catastrophic one here.